tuist/tuist
# Ensure test results are cached when specifying a scheme to test

### Short description 📝

I noticed recently that tests I was running were not getting cached in the `~/.tuist/Cache/TestsCache` directory as expected. After some help from @fortmarek and some experimentation, I determined that tests _do_ get cached, but _only_ if running `tuist test` without specifying a scheme. If a scheme is specified (`tuist test MyScheme`), then the test results are not cached.

Turns out the fix was pretty simple. There's an `if` statement that checks if a scheme was specified or not, and the code that saves the test results to the cache was only being run in the `else` part of the statement — when there was no scheme specified. The change is to move the code after the `if` statement so that it's run either way. (Thanks to @fortmarek on the advice on how to address this.)

### How to test the changes locally 🧐

- Run `tuist clean` to clean out your local `~/.tuist` directory.
- On a Tuist repo, run `tuist test <scheme>`, and verify that the test results are cached in `~/.tuist/Cache/TestsCache`.

### Checklist ✅

- ~~In case the PR introduces changes that affect users, the documentation has been updated.~~
- ~~In case the PR introduces changes that affect how the cache artifact is generated starting from the `TuistGraph.Target`, the `Constants.cacheVersion` has been updated.~~
